File: test_integracion/FunkoShop_Exam/src/main/java/es/codeurjc/test/exam/FunkoShop/FunkoShop.java
package es.codeurjc.test.exam.FunkoShop;

import java.util.Optional;

public class FunkoShop {

    private FunkoRepository funkoRepository;
    
    public FunkoShop(FunkoRepository funkoRepository){
        this.funkoRepository = funkoRepository;
    }

    public Funko buy(String funkoName){
        Optional<Funko> optional = funkoRepository.findByName(funkoName);
        if(optional.isPresent()){
            Funko funko = optional.get();
            if(funko.isAvailable()){
                funko.setAvailable(false);
                funkoRepository.save(funko);
                return funko;
            }else{
                throw new RuntimeException("Funko not available");
            }
        }else{
            throw new RuntimeException("Funko not found");
        }
    }
}

File: test_integracion/FunkoShop_Exam/src/FunkoShop.js
class FunkoShop {
    constructor(funkoRepository) {
        this.funkoRepository = funkoRepository;
    }

    async buy(funkoName) {
        const funko = await this.funkoRepository.findByName(funkoName);
        if (!funko) {
            throw new Error('Funko not found');
        }
        if (!funko.isAvailable) {
            throw new Error('Funko not available');
        }
        funko.isAvailable = false;
        this.funkoRepository.save(funko);
        return funko;
    }
}

module.exports = { FunkoShop };